### DISCUSSION TITLE
First time flashing - getting Failed to connect to ESP8266: tuned out waiting for packet handler

### DISCUSSION THREAD

**Discussion by wayner9** [arendst/Tasmota#19753] [Q&A]
I am trying to flash some devices to Tasmota for the first time.  I am using Tasmotizer. I tried both a Sonoff Basic R2 and a Sonoff POWR3 which has header pins.  I have reversed the Tx and Rx to see if that was wrong.  I have tried with two different PCs.  Yet I always get the same error.

Any suggestions?  I am guessing that maybe my USB to serial TTL device doesn't work.  I bought a pair of these from Amazon https://www.amazon.ca/dp/B01JG8H5U4?psc=1&ref=ppx_yo2ov_dt_b_product_details

**Gikfun FT232RL 3.3V 5V FTDI USB to TTL Serial Adapter Module for Arduino Mini Port (Pack of 2pcs) AE1186x2** 

This device supports both 3.3V and 5V.  I have the jumper across 3.3V and I have verified the voltage with a voltmeter.

I hold down the button on the devices for 5+ seconds when I plug in the USB cable.  A COM port shows up in Tasmotizer, but I always get this error.

I am trying to save the original firmware, using Release 13.1.0, erase before flash is checked and using a flash size of 1MB.

**Comment by sfromis** [comment]
That USB/serial adapter cannot supply power for reliable flashing. It may "often" work for ESP8266 devices, but is still a notable risk factor, instead of the [recommended](https://tasmota.github.io/docs/Getting-Started/#serial-programmer) ones. If you have a good 3.3V power source on the side, you could try using that for power, and of course common ground.

Getting "timed out waiting for packet header" can have several causes, as it basically indicate some communication issue. Not having good power for flashing is one possibility, but also things like other programs using the serial port, going via a USB hub, cable issues, or not being in flash mode are not excluded.

**Reply by wayner9** [reply]
Thank you, I didn't pay too much attention initially, I thought that the USB-Serial adapter didn't matter too much. I will get one of the CH340G ones.

In the meantime is there any other way to supply the 3.3V of power?  Can I just use a couple of AA batteries?  A pair of fresh alkaline batteries give about 3.3V.

**Reply by sfromis** [reply]
As I already suggested, if you have a good 3.3V power source, it should be workable, which would include batteries. Exact voltage is not highly critical, but should be well within +/- 10%.

The reason for the USB/serial adapter mattering a lot is that many of those are simply not meant to also act as a 3.3V power supply for things like running an ESP8266 device. In general, you should use one with an extra voltage regulator on the board. At least 150 mA recommended.

**Comment by Jibun-no-Kage** [comment]
You could use an AS1117 module, to down step 5v to 3.3v as power source, of course with common ground.  This lets me use a separate 5v USB PS.
